infra/resources/prod/function_profile.tf

```
@@ -0,0 +1,230 @@
+      ON_PROFILE_UPDATE_LEASES_PREFIX         = "OnProfileUpdateLeasesPrefix-001"
+
+      MAILUP_USERNAME = data.azurerm_key_vault_secret.common_MAILUP_USERNAME.value
+      MAILUP_SECRET   = data.azurerm_key_vault_secret.common_MAILUP_SECRET.value
+      PUBLIC_API_KEY  = trimspace(data.azurerm_key_vault_secret.fn_app_PUBLIC_API_KEY.value)
+      // TODO: probably does not needed anymore due the fn which was using it is now moved in io-profile-async
+      SPID_LOGS_PUBLIC_KEY = trimspace(data.azurerm_key_vault_secret.fn_app_SPID_LOGS_PUBLIC_KEY.value)
+      AZURE_NH_ENDPOINT    = data.azurerm_key_vault_secret.fn_app_AZURE_NH_ENDPOINT.value // TODO: searching this config in the codebase show no results
+
+    }
+  }
+}
+
+
+module "function_profile" {
+  source  = "pagopa-dx/azure-function-app/azurerm"
+  version = "~> 0.0"
+
+  environment = {
+    prefix          = local.prefix
+    env_short       = local.env_short
+    location        = local.location
+    domain          = local.domain
+    app_name        = local.function_profile.name
+    instance_number = "02"
+  }
+
+  node_version          = 20
+  health_check_path     = "/api/v1/info"
+  has_durable_functions = "true"
+
+  # P2mv3 SKU and 8 Worker process count(sku set as of io-infra function-profile definition)
+  tier = "xl"
+
+
+  resource_group_name = azurerm_resource_group.main_resource_group.name
+
+  subnet_cidr   = local.cidr_subnet_fn_profile
+  subnet_pep_id = data.azurerm_subnet.private_endpoints_subnet.id
+
+  private_dns_zone_resource_group_name = data.azurerm_resource_group.rg_common.name
+
+  virtual_network = {
+    name                = data.azurerm_virtual_network.itn_common.name
+    resource_group_name = data.azurerm_virtual_network.itn_common.resource_group_name
+  }
+
+  app_settings      = local.function_profile.app_settings
+  slot_app_settings = local.function_profile.app_settings
+
+  subnet_service_endpoints = {
+    web = true
+  }
+
+  application_insights_connection_string = data.azurerm_application_insights.application_insights.connection_string
+
+  action_group_id = azurerm_monitor_action_group.error_action_group.id
+
+  tags = local.tags
+}
+
+module "function_profile_autoscale" {
+  depends_on = [module.function_profile]
+  source     = "pagopa/dx-azure-app-service-plan-autoscaler/azurerm"
+  // TODO: in order to update to version 1.0.0, add the required inputs `app_service_plan_id` and `location`
+  version = "~> 0.0"
+
+  resource_group_name = azurerm_resource_group.main_resource_group.name
+  target_service = {
+    function_app_name = module.function_profile.function_app.function_app.name
+  }
+
+  scheduler = {
+    normal_load = {
+      minimum = 3
+      default = 10
+    },
+    maximum = 30
+  }
+
+  scale_metrics = {
+    requests = {
+      statistic_increase        = "Max"
+      time_window_increase      = 1
+      time_aggregation          = "Maximum"
+      upper_threshold           = 2500
```

actual configuration of the autoscale settings includes `2000` as upper threshold
```suggestion
      upper_threshold           = 2000
```

---

yup, aligned as io-infra definition in e2ce1942e0d4132d9b9d5e03d40c66c155fb8a9b